src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -1038,7 +1071,7 @@ namespace smt::noodler {
         prep_handler.propagate_variables();
         prep_handler.propagate_eps();
         prep_handler.infer_alignment();
-        prep_handler.remove_regular();
+        prep_handler.remove_regular(conv_vars);
```

Comment:
If we find out that conversion variables are necessary also for other preprocessing rules, we should pass them in constructor.